Mathlib/Data/Set/Finite/Basic.lean

```diff
@@ -688,6 +688,22 @@ theorem Finite.dinduction_on {C : ∀ s : Set α, s.Finite → Prop} (s : Set α
     Finite.induction_on h (fun _ => H0) fun has hs ih _ => H1 has hs (ih _)
   this h
 
+theorem setOf_nodup_perm (l : List α) (H : l.Nodup) [DecidableEq α] :
+    l.finite_toSet.toFinset.toList.Perm l := by
```

Comment:
I think this is almost the same as [`List.toFinset_toList`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Dedup.html#List.toFinset_toList)?

Reply:
Indeed, it seems the missing piece is `{x | x ∈ l}.toFinset = l.toFinset`, which would be a good simp lemma.

Reply:
Why are we writing `{x | x ∈ l}` to begin with? We could be writing `↑l.toFinset` directly.

Reply:
> `↑l.toFinset` directly.

Only with `DecidableEq`.